src/Tuura/SelfDual.hs

```diff
@@ -0,0 +1,56 @@
+module Tuura.SelfDual (
+    isSelfDual, getSelfDuals) where
+
+import Data.List (sort)
+import Control.Monad (replicateM)
+
+import Tuura.Boolean
+
+isSelfDual :: String -> Bool
```

Comment:
Let's separate two concerns: (1) parsing a function from a `String` and (2) testing that the function is self-dual. It would make (2) easier to reuse.